terraform/modules/data-share-service/ec2.tf

```diff
@@ -0,0 +1,75 @@
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
+  }
+
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical
+}
+
+resource "aws_network_interface" "rds_bastion_eni" {
+  subnet_id       = module.vpc.public_subnet_ids[0]
+  security_groups = [aws_security_group.rds_bastion_host_sg.id]
+  attachment {
+    device_index = 0
+    instance     = aws_instance.rds_bastion_host.id
+  }
+}
+
+resource "aws_instance" "rds_bastion_host" {
+  ami                         = data.aws_ami.ubuntu.id
+  instance_type               = "t4g.nano"
+  associate_public_ip_address = true
+  key_name                    = aws_key_pair.rds_bastion_key_pair.key_name
+  metadata_options {
+    http_endpoint = "disabled"
+  }
+  root_block_device {
+    encrypted = true
+  }
+}
+
+resource "aws_key_pair" "rds_bastion_key_pair" {
+  public_key = module.tls_private_key.public_key_pem
+  key_name   = "rds-bastion-key-pair-${var.environment}"
+}
+
+#tfsec:ignore:aws-ec2-no-public-ingress-sgr
```

Comment:
Is there any sensible limit we can put on this?

Reply:
We could use an IP allow list? I don't know what values we would use other than the SW IP ranges though

Reply:
for the moment this is good, but could we add a TODO

Reply:
Created ticket